### PR TITLE
Fix duplicated class names

### PR DESCRIPTION
**Pull request by valerio-iachini**
This PR fixes a bug that leads to the generation of duplicated codes when multiple JSON schema definitions share the same property name. As a result, the code generation process produces multiple classes whit the same name.

### CODE REVIEW

File: packages/org.json_schema.contrib/internal/ModuleGenerator.pkl [hunk 1]
```diff
@@ -222,13 +222,14 @@ function isClassLike(schema: JsonSchema.Schema): Boolean =
 /// Try to use the parent property's name as part of the class name in case of conflict.
 /// If already at the root, add a number at the end.
 local function determineTypeName(path: List<String>, candidateName: String, existingTypeNames: Set<Type>, index: Int): Type =
-  if (existingTypeNames.contains(utils.pascalCase(candidateName)))
-    if (path.isEmpty)
-      determineTypeName(path, candidateName + index.toString(), existingTypeNames, index + 1)
+  let (candidateType = new Type { name = utils.pascalCase(candidateName); moduleName = module.moduleName })
+    if (existingTypeNames.contains(candidateType))
```
Comment (holzensp) [inline packages/org.json_schema.contrib/internal/ModuleGenerator.pkl:226]: This can't be right, can it?
```suggestion
    if (existingTypeNames.contains(candidateType.name))
```

Could you add a test for the case that you're fixing?

**Review by holzensp — CHANGES_REQUESTED** [review]
Thanks for your contribution. I _believe_ I spotted a bug. If I'm right, could you make sure that bug is covered in the tests?

File: packages/org.json_schema.contrib/internal/ModuleGenerator.pkl (hunk 1 above)
Comment (valerio-iachini) [reply packages/org.json_schema.contrib/internal/ModuleGenerator.pkl:226]: Yes, I'll add tests. `existingTypeNames` contains a list of `Type`, so if we call `contains` with only the name, it always returns `false`. The `determineTypeName` function returns a `Type` that is added to the accumulator, which is passed again to the function. The name `typeNames` is also used because there is an alias [here](https://github.com/apple/pkl-pantry/blob/4f8f939c35ff1fdc04b0897db4b53c05ff234186/packages/org.json_schema.contrib/internal/Type.pkl#L25) that states that a map of JSON schema and `Type` is called `TypesName`.

Comment (valerio-iachini) [reply packages/org.json_schema.contrib/internal/ModuleGenerator.pkl:226]: The code that I fixed with this PR incorrectly called the `contains` method with just the name, causing it to consistently return `false`. This resulted in the generation of classes with the same name when encountering properties of type object with identical names in the schema.

File: packages/org.json_schema.contrib/internal/ModuleGenerator.pkl [hunk 2]
```diff
@@ -240,7 +241,7 @@ local classSchemas: Type.TypeNames =
       .fold(Map(), (accumulator: Type.TypeNames, pair) ->
         let (path = pair.first)
         let (schema = pair.second)
-        let (typeName = determineTypeName(path, path.lastOrNull?.capitalize() ?? "Item", accumulator.values.toSet(), 0))
+        let (typeName = determineTypeName(path.dropLast(1), path.lastOrNull?.capitalize() ?? "Item", accumulator.values.toSet(), 0))
```
Comment (tbinna) [inline packages/org.json_schema.contrib/internal/ModuleGenerator.pkl:244]: Could you explain why this change is necessary?

File: packages/org.json_schema.contrib/internal/ModuleGenerator.pkl (hunk 1 above)
Comment (tbinna) [reply packages/org.json_schema.contrib/internal/ModuleGenerator.pkl:226]: The implementation here may be a more straightforward solution: https://github.com/apple/pkl-pantry/pull/21/files

Comment (valerio-iachini) [reply packages/org.json_schema.contrib/internal/ModuleGenerator.pkl:226]: Yeah, I see what you mean, but it's not exactly the same. Is it acceptable to have a type with the same name in a different module? Also, every time the function is called, it remaps all the types to their corresponding names. Additionally, this PR addresses another issue: the generated classes have the property name repeated twice.

File: packages/org.json_schema.contrib/internal/ModuleGenerator.pkl (hunk 2 above)
Comment (valerio-iachini) [reply packages/org.json_schema.contrib/internal/ModuleGenerator.pkl:244]: Otherwise, the generated classes end up with the property name repeated twice. You can check out the code snippet [here](https://github.com/apple/pkl-pantry/blob/be26d4171c97b5dca365a90a6397ff2739c5933c/packages/org.json_schema.contrib/internal/ModuleGenerator.pkl#L230C42-L230C52), where the first argument is appended with the current candidate name.

Comment (tbinna) [reply packages/org.json_schema.contrib/internal/ModuleGenerator.pkl:244]: Ah yes, I can see this in my output now too. Thanks for clarifying

File: packages/org.json_schema.contrib/internal/ModuleGenerator.pkl (hunk 1 above)
Comment (holzensp) [reply packages/org.json_schema.contrib/internal/ModuleGenerator.pkl:226]: You are right; I didn't follow through on the actual definition. In this case, please rename `s/existingTypeNames/exstingTypes/`, because it's pretty misleading currently. I don't want you to be blocked on that, though, because it's PR-scope-creep.

**Review by holzensp — APPROVED** [review]
Happy to take this as is (when CI is green); happy for you to roll in the rename.

**Comment by tbinna** [comment]
Fixes #20
